joblib/numpy_pickle.py

```diff
@@ -226,75 +233,75 @@ def __init__(self, filename, compress=0, cache_size=10, protocol=None):
             np = None
         self.np = np
 
-    def _write_array(self, array, filename):
-        if not self.compress:
-            self.np.save(filename, array)
-            allow_mmap = not array.dtype.hasobject
-            container = NDArrayWrapper(os.path.basename(filename),
-                                       type(array),
-                                       allow_mmap=allow_mmap)
-        else:
-            filename += '.z'
-            # Efficient compressed storage:
-            # The meta data is stored in the container, and the core
-            # numerics in a z-file
-            _, init_args, state = array.__reduce__()
-            # the last entry of 'state' is the data itself
-            with open(filename, 'wb') as zfile:
-                write_zfile(zfile, state[-1], compress=self.compress)
-            state = state[:-1]
-            container = ZNDArrayWrapper(os.path.basename(filename),
-                                            init_args, state)
-        return container, filename
+    def _create_array_wrapper(self, array):
+        """Create and returns a numpy array wrapper from a numpy array."""
+        order = 'F' if (array.flags.f_contiguous and
+                        not array.flags.c_contiguous) else 'C'
+        allow_mmap = not self.buffered and not array.dtype.hasobject
+        wrapper = NumpyArrayWrapper(type(array),
+                                    array.shape, order, array.dtype,
+                                    allow_mmap=allow_mmap)
+
+        return wrapper
 
     def save(self, obj):
-        """ Subclass the save method, to save ndarray subclasses in npy
-            files, rather than pickling them. Of course, this is a
-            total abuse of the Pickler class.
+        """Subclass the Pickler `save` method.
+
+        This is a total abuse of the Pickler class in order to use the numpy
+        persistence function `save` instead of the default pickle
+        implementation. The numpy array is replaced by a custom wrapper in the
+        pickle persistence stack and the serialized array is written right
+        after in the file. Warning: this breaks the pickle format and prevents
```

Comment:
Rather than "this breaks ...", "the file produced does not follow the pickle format. As such it can not be read with pickle.load."